File: EatSomewhere/Data/Ingredient.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using EatSomewhere.Users;

namespace EatSomewhere.Data;

public class Ingredient
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public string Id { get; set; }
    public User CreatedBy { get; set; }
    public string Name { get; set; }
    /// <summary>
    /// Cents
    /// </summary>
    public int Cost { get; set; }
    public double Amount { get; set; }
    public Unit Unit { get; set; }
    public Assembly Assembly { get; set; }
    public bool Archived { get; set; } = false;

    [JsonIgnore] public List<IngredientEntry> IngredientEntries { get; set; } = new();
}

File: app/lib/backend_data/ingredient.dart
import 'package:eat_somewhere/backend_data/assembly.dart';

import 'unit.dart';

class Ingredient {
  String Id;
  String Name;
  double Cost;
  double Amount;
  IngredientUnit Unit;
  DummyAssembly Assembly;

  Ingredient({
    required this.Id,
    required this.Name,
    required this.Cost,
    required this.Amount,
    required this.Unit,
    required this.Assembly,
  });

  factory Ingredient.fromJson(Map<String, dynamic> json) {
    return Ingredient(
      Id: json['Id'],
      Name: json['Name'],
      Cost: json['Cost'],
      Amount: json['Amount'],
      Unit: IngredientUnit.values[json['Unit']],
      Assembly: DummyAssembly.fromJson(json['Assembly']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'Id': Id,
      'Name': Name,
      'Cost': Cost,
      'Amount': Amount,
      'Unit': Unit.index,
      'Assembly': Assembly.toJson(),
    };
  }
}
